csrc/host_ir/lower_to_llvm.h

```diff
@@ -0,0 +1,143 @@
+
+
+#include <fusion.h>
+#include <global_allocator.h>
+#include <host_ir/container.h>
+#include <host_ir/executor.h>
+#include <ir/all_nodes.h>
+#include <ops/all_ops.h>
+#include <val_graph_visitor.h>
+
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/IR/Verifier.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/IR/LLVMContext.h"
+#include <llvm/ExecutionEngine/Orc/ThreadSafeModule.h>
+#include <llvm/ExecutionEngine/Orc/LLJIT.h>
+#include <llvm/ExecutionEngine/Orc/CompileUtils.h>
+#include <llvm/ExecutionEngine/Orc/IRCompileLayer.h>
+#include <llvm/ExecutionEngine/Orc/RTDyldObjectLinkingLayer.h>
+#include <llvm/ExecutionEngine/JITLink/JITLink.h>
+#include "llvm/Support/Error.h"
+
+#include <unordered_map>
+#include <queue>
+
+
+namespace nvfuser {
+
+/*
+
+Helper Data Structures & Functions
+
+*/
+
+// Codegen type for the shape inference
+enum class codegenType{
+  Merge, // nvfuser merge -> llvm mul
+  Split // nvfuser split -> llvm constant padding + udiv
+};
+
+// Dependency graph entry for the shape inference
+class dependency_graph{
```

Comment:
IIUC, you just wanted a topo order traversal?

We should revisit this... I think we have utils for sorting already.

Reply:
If we don't, we should add that :wink: